### PR TITLE
EUI-8904 CR-84 Hearings semi-automatic updates

### PR DESCRIPTION
**Pull request by johnbenjamin-hmcts**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/EUI-8904

### Change description ###
CR-84 Tech Ticket - Hearings manual update - Semi-automatically updated

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

**Review by Daniel-Lam — APPROVED** [review]
@johnbenjamin-hmcts - looks OK; approved

**Review by DavidJayakumar — APPROVED** [review]

File: src/hearings/containers/request-hearing/hearing-view-edit-summary/hearing-view-edit-summary.component.ts [hunk 1]
```diff
@@ -64,7 +66,7 @@ export class HearingViewEditSummaryComponent extends RequestHearingPageFlow impl
   }
 
   public setPropertiesUpdatedOnPageVisit(): void {
-    this.hearingValuesSubscription = this.hearingStore.select(fromHearingStore.getHearingValues).pipe(take(1)).subscribe((hearingValues) => {
+    this.hearingValuesSubscription = this.hearingStore.select(fromHearingStore.getHearingValues).subscribe((hearingValues) => {
```
Comment (Daniel-Lam) [inline src/hearings/containers/request-hearing/hearing-view-edit-summary/hearing-view-edit-summary.component.ts:69]: @johnbenjamin-hmcts - now that `this.hearingValuesSubscription` has had the `take(1)` removed, is this `Subscription` being unsubscribed somewhere?

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@johnbenjamin-hmcts - just one query that needs reviewing, otherwise all OK

File: src/hearings/containers/request-hearing/hearing-view-edit-summary/hearing-view-edit-summary.component.ts (hunk 1 above)
Comment (johnbenjamin-hmcts) [reply src/hearings/containers/request-hearing/hearing-view-edit-summary/hearing-view-edit-summary.component.ts:69]: @Daniel-Lam - the `this.hearingValuesSubscription` is unsubscribed in the `ngOnDestroy()` method.

The `take(1)` in this case was not returning the latest results, whereas returning the previous results from the store, so had to remove it.

**Review by Daniel-Lam — APPROVED** [review]

File: src/cases/components/case-hearings-list/case-hearings-list.component.ts [hunk 2]
```diff
@@ -137,4 +137,9 @@ export class CaseHearingsListComponent implements OnInit {
   public LoadHearingRequestAndRedirect(hearingID: string, targetURL: string) {
     this.hearingStore.dispatch(new fromHearingStore.LoadHearingRequest({ hearingID, targetURL }));
   }
+
+  public LoadHearingRequestServiceHearingValuesAndRedirect(hearingID: string, targetURL: string) {
```
Comment (Daniel-Lam) [inline src/cases/components/case-hearings-list/case-hearings-list.component.ts:141]: @johnbenjamin-hmcts - how come the function name begins with a capital letter? I've not seen that often... if at all :) Same for the one on Line 137

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@johnbenjamin-hmcts - just a minor observation; everything else looks OK

File: src/cases/components/case-hearings-list/case-hearings-list.component.ts (hunk 2 above)
Comment (johnbenjamin-hmcts) [reply src/cases/components/case-hearings-list/case-hearings-list.component.ts]: @Daniel-Lam - thanks for identifying this. I don't know why `LoadHearingRequestAndRedirect` was coded to start with a capital letter, and when I copied and created a new method `LoadHearingRequestServiceHearingValuesAndRedirect` didn't change it as well.

I have now modified both of the methods to start with a capital letter.

**Review by DavidJayakumar — APPROVED** [review]
LGTM

**Pull request merged**